src/controllers/menu.controller.ts

```diff
@@ -0,0 +1,82 @@
+import * as Koa from 'koa'
+import {getRepository, Repository} from 'typeorm'
+import Dish from '../models/menu.entity'
+import * as HttpStatus from 'http-status-codes'
+import {Like} from "typeorm";
+
+export default class MenuController {
+
+    static async getMenu(ctx: Koa.Context): Promise<void> {
+        const menuRepo: Repository<Dish> = getRepository(Dish)
+        const DishName = ctx.request.query.name
+        const CategoryName = ctx.request.query.category
+        let dishes: Dish[]
+        if (ctx.request.query.name) {
+            dishes = await menuRepo.find(
+                {
+                    name: Like(`%${(DishName as string).toUpperCase()}%`)
+                }
+            )
+        } else if (ctx.request.query.category) {
+            dishes = await menuRepo.find(
+                {
+                    where: {
+                        dishCategory: (CategoryName as string).charAt(0).toUpperCase() + (CategoryName as string).slice(1).toLowerCase()
```

Comment:
better move it to function like `capitalize` or even use some library like lodash for it